x-pack/plugins/security_solution/public/resolver/view/panels/panel_content_related_counts.tsx

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiI18nNumber,
+  EuiBasicTableColumn,
+  EuiButtonEmpty,
+  EuiSpacer,
+  EuiInMemoryTable,
+} from '@elastic/eui';
+import { CrumbInfo, StyledBreadcrumbs } from './panel_content_utilities';
+
+import * as event from '../../../../common/endpoint/models/event';
+import { ResolverEvent, ResolverNodeStats } from '../../../../common/endpoint/types';
+
+/**
+ * This view gives counts for all the related events of a process grouped by related event type.
+ * It should look something like:
+ *
+ * | Count                  | Event Type                 |
+ * | :--------------------- | :------------------------- |
+ * | 5                      | DNS                        |
+ * | 12                     | Registry                   |
+ * | 2                      | Network                    |
+ *
+ */
+export const EventCountsForProcess = memo(function EventCountsForProcess({
+  processEvent,
+  pushToQueryParams,
+  relatedStats,
+}: {
+  processEvent: ResolverEvent;
+  pushToQueryParams: (arg0: CrumbInfo) => unknown;
+  relatedStats: ResolverNodeStats;
+}) {
+  interface EventCountsTableView {
+    name: string;
+    count: number;
+  }
+
+  const relatedEventsState = { stats: relatedStats.events.byCategory };
+  const processName = processEvent && event.eventName(processEvent);
+  const processEntityId = event.entityId(processEvent);
+  const totalCount = Object.values(relatedStats.events.byCategory).reduce(

```

Comment:
I know we've talked about how the "total" for related events might add up differently than the values in the map so I'm not sure if that's what we're trying to address here. Just in case though I wanted to point out that `ResolverNodeStats` has a total with it:

https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/common/endpoint/types.ts#L48

`relatedStats.events.total`

That indicates how many related event documents there are

Reply:
answered in another thread: we're showing the total _aggregate counts_ to prevent confusing the user, may adjust approach later based on user feedback